nats.go

```diff
@@ -2559,6 +2572,28 @@ func (nc *Conn) stopPingTimer() {
 	}
 }
 
+// DefaultReconnectBackoffHandler returns a default reconnect exponential backoff interval.
+// Base reconnect wait is 10ms, with x2 multiplier. Max wait time is 2 minutes.
+// 10ms, 20ms, 40ms, 80ms...2m
+// A random jitter is added to the result.
+func DefaultReconnectBackoffHandler(jitter time.Duration) ReconnectDelayHandler {
```

Comment:
With this feature, I am wondering if some of the other options we have on the clients should be deprecated or go away (at the very least, this should be the default)